### ISSUE TITLE
thread-safty?

### ISSUE THREAD

**Issue by scottzhang** [bacnet-stack/bacnet-stack#810]
I am trying to integrate bacnet-stack into my c++ program which dynamic load so drivers into my program. 

I saw many static global variable in c files, eg:
static union {
    error_function error;
    complex_error_function complex;
} Error_Function[MAX_BACNET_CONFIRMED_SERVICE];

Base on my understanding, these variable, when so is being loaded into program address, will only have single copy. which means if I create several client to request data from peers, their actions may conflicts. 

So I think many code need to be fixed before multi-threading cases use

**Comment by skarg** [comment]
Yes, there are 'singletons' in the library - specifically in the 'src/bacnet/basic/' tree.  You are welcome to create a src/bacnet/??? to support a different approach while still using the core encoding, decoding, and ports that you need.
